services/bankid-api/lambdas/sign.js

```diff
@@ -1,42 +1,68 @@
+import to from 'await-to-js';
 import { failure, success } from '../../../libs/response';
+import { validateEventBody } from '../../../libs/validateEventBody';
+import { validateKeys } from '../../../libs/validateKeys';
+import { throwError } from '@helsingborg-stad/npm-api-error-handling';
 import params from '../../../libs/params';
+import config from '../../../config';
 import * as request from '../../../libs/request';
 import * as bankId from '../helpers/bankId';
 
-const SSMParams = params.read('/bankidEnvs/dev');
+const SSMParams = params.read(config.bankId.envsKeyName);
+let valid = true;
 
-export const main = async event => {
-  try {
-    const bankidSSMParams = await SSMParams;
-    const { endUserIp, personalNumber, userVisibleData } = JSON.parse(event.body);
-
-    const payload = {
-      endUserIp,
-      personalNumber,
-      userVisibleData: userVisibleData
-        ? Buffer.from(userVisibleData).toString('base64') // eslint-disable-line no-undef
-        : undefined,
-    };
-
-    const bankIdClient = await bankId.client(bankidSSMParams);
-
-    const { data } = await request.call(
-      bankIdClient,
-      'post',
-      bankId.url(bankidSSMParams.apiUrl, '/sign'),
-      payload
-    );
-
-    const { orderRef, autoStartToken } = data;
-
-    return success({
-      type: 'bankIdSign',
-      attributes: {
-        order_ref: orderRef,
-        auto_start_token: autoStartToken,
-      },
-    });
-  } catch (error) {
-    return failure({ status: false, error: error.message });
+// Not validating personalNumber as it is an optional Parameter
+function validateTokenEventBody(body) {
+  let errorStatusCode, errorMessage;
+  valid = true;
+
+  if (!validateKeys(JSON.parse(body), ['endUserIp', 'userVisibleData'])) {
+    valid = false;
+    errorStatusCode = 400;
+    errorMessage = 'Missing JSON body parameter';
   }
+
+  return [valid, errorStatusCode, errorMessage];
+}
+
+async function sendBankIdSignRequest(params, payload) {
+  let error, bankIdClientResponse, bankIdSignResponse;
+  [error, bankIdClientResponse] = await to(bankId.client(params));
+
+  if (!bankIdClientResponse) throwError(error.response.status, error.response.data.details);
+
+  [error, bankIdSignResponse] = await to(
+    request.call(bankIdClientResponse, 'post', bankId.url(params.apiUrl, '/sign'), payload)
+  );
+
+  if (!bankIdSignResponse) throwError(error.response.status, error.response.data.details);
+  return bankIdSignResponse;
+}
+
+export const main = async event => {
```

Comment:
A minor thing. But we have the export of Main functions at the top in all our other services. So to be consistent i think se should move this to the top aswell. 

However i don't know if this is better acctually. Feels more natural to export at the end of the file 😄

Reply:
Normally I keep the export function in the end and sub function on top. :)

But I changed it to top for consistency.

Reply:
Yeah maybe we should discuss the export syntax (y)